src/robot/libraries/dialogs_ipy.py

```diff
@@ -12,32 +12,183 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 
+import wpf
+from System.Windows import Application, GridLength, SizeToContent, TextWrapping, \
+                           Thickness, Window, WindowStartupLocation, WindowState
+from System.Windows.Controls import Button, ColumnDefinition, Grid, Label, ListBox, \
+                                    PasswordBox, RowDefinition, TextBlock, TextBox
 
-class _AbstractWinformsDialog:
 
+class _mainWindow(Window):
```

Comment:
We can use any naming you prefer.

This is critical, because without the Application.MainWindow defined the
Dialog itself becomes the MainWIndow and we cannot run multiple
applications in the same AppDomain. In other words, only the first 'atest'
will run, and the rest will all fail with an error about the AppDomain.

On Wed, Sep 2, 2015 at 1:25 AM, Pekka Klärck <notifications@github.com>
wrote:

> In src/robot/libraries/dialogs_ipy.py
> <https://github.com/robotframework/robotframework/pull/2126#discussion_r38508591>
> :
>
> >
> > +class _mainWindow(Window):
>
> Should be named _MainWindow or just MainWindow. But is this really
> needed? This seems to be used only once and that usage could perhaps be
> inlined.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/robotframework/robotframework/pull/2126/files#r38508591>
> .
>